### PR TITLE
Drop support for '--no-minibrowser' mode

### LINKED ISSUE #33626: Drop support for '--no-minibrowser' mode in servoshell
The `--no-minibrowser` mode was added as fallback option when we first added the mini-browser mode, to be used in case the mini-browser broke servoshell.

It is currently not very well tested and has few issues that remain open ([#33254](https://github.com/servo/servo/issues/33254) and [33255](https://github.com/servo/servo/issues/33255)). 

Supporting both modes in servoshell also makes it harder to extend and maintain. For example, one of the planned tasks for our Outreachy project is to move away from 'tinyfiledialog-rs' to something modern like [egui_modal](https://crates.io/crates/egui-modal). However, `egui_modal` requires an egui `Context` to work, which is not available in --no-minibrowser mode.

Given these reasons, it seems like a good time to drop support for `--no-minibrowser` mode from servoshell.

This will involve removing:
1. The CLI argument in https://github.com/servo/servo/blob/c7a4e4f627f0bf331cedc1700ddeb92ffcba4e78/components/config/opts.rs#L567 and config option in https://github.com/servo/servo/blob/c7a4e4f627f0bf331cedc1700ddeb92ffcba4e78/components/config/opts.rs#L133
2. Remove or update the conditional logic under `ports/servoshell/desktop` folder that uses the `opts::get().minibrowser`.

### PR DESCRIPTION
**Pull request by Taym95**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #33626

- [x]  these changes do not require tests because changes were made to ServoShell.

### CODE REVIEW

File: ports/servoshell/desktop/webview.rs [hunk 1]
```diff
@@ -431,29 +431,7 @@ where
...
-                        location_bar_input_to_url(&input).and_then(|url| {
-                            self.focused_webview_id
-                                .map(|id| EmbedderEvent::LoadUrl(id, url))
-                        })
-                    })
-                } else {
-                    None
-                }
-            })
+            .shortcut(CMD_OR_CONTROL, 'L', || None)
```
Comment (mukilan) [inline ports/servoshell/desktop/webview.rs:434]: I think we can simply remove this call to shortcut as it doesn't have any effect.

File: ports/servoshell/desktop/webview.rs [hunk 2]
```diff
@@ -431,29 +431,7 @@ where
...
-                            .as_ref()
-                            .map(|url| url.to_string())
-                            .unwrap_or_else(String::default),
-                        None => String::default(),
-                    };
-
-                    let title = "URL or search query";
-                    let input = tinyfiledialogs::input_box(title, title, &tiny_dialog_escape(&url));
-                    input.and_then(|input| {
-                        location_bar_input_to_url(&input).and_then(|url| {
```
Comment (mukilan) [inline ports/servoshell/desktop/webview.rs:448]: This now causes "unused import" warning for `location_bar_input_to_url`. Could you remove the corresponding `use` statement as well.

**Review by mukilan — CHANGES_REQUESTED** [review]

File: ports/servoshell/desktop/webview.rs (hunk 1 above)
Comment (Taym95) [reply ports/servoshell/desktop/webview.rs]: done

File: ports/servoshell/desktop/webview.rs (hunk 2 above)
Comment (Taym95) [reply ports/servoshell/desktop/webview.rs:448]: done

**Review by mukilan — APPROVED** [review]
Thank you!

**Pull request merged**